qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py

```diff
@@ -217,60 +181,60 @@ def handle_calculate_phase_table_clicked(self):
         if name is None:
             self.enable_editing_notifier.notify_subscribers()
             return
-        self._new_table_name = name
+        self.model.new_table_name = name
 
         # Calculate the new table in a separate thread
-        self.calculation_thread = self.create_phase_table_calculation_thread()
-        self.calculation_thread.threadWrapperSetUp(
+        self.model.calculation_thread = self.create_phase_table_calculation_thread()
```

Comment:
I'm wondering if data calculations like this one and the phasequad calculation on line 275 might be better in the model, assuming they work OK there?

Reply:
I think I left it there so it would be able to easily add the `self.handle_phase_table_calculation*` methods.

Reply:
Yes, I guess I was thinking that technically you might be able to pass the callback methods through to the model in the same way that they're currently being passed to a method on the calculation thread. Maybe this is one to leave for Rob's advice, though, if we ask him to gatekeeper this.

Reply:
@robertapplin what do you think?

Reply:
I think Rachel's idea sounds good, to pass the callback methods through to the model, assuming its simple enough to do this. But if its a lot of work, then its not so bad to keep it the same

Reply:
Thanks @robertapplin, I've had a go at doing that. I think you could probably open an issue to look at adding a bit more test coverage to the new model, but hopefully this will do for now.

@jhaigh0 I'm happy with all the changes you've made on this. Once all the tests have run through, perhaps you could review just the bits I've refactored in the last commit? If that all looks OK to you then I'll approve this and it can be gatekeepered. I've run through the manual tests since my refactor to check it's working OK, and all looks fine to me.